### PR TITLE
Added recursive switch to parser and hooked up recursive file search

### PR DESCRIPTION
**Pull request by reggiemcdonald**
✅ New `-r|--recursive` switch
✅ Hooked up the recursive file search with the UI

### CODE REVIEW

File: src/File.hs [hunk 1]
```diff
@@ -29,6 +28,19 @@ data ExFile = ExFile {
...
 
+{-|
+    @brief Given a filepath and a recursive flag, dispatches
+    the correct file finder.
+    @throws IOException if file/dir does not exist
+    @param filepath String
+    @param recursive Bool
+    @returns list of files and their contents
+-}
+dispatchFileExtraction :: FilePath -> Bool -> IO [ExFile]
```
Comment (Julian-Mentasti) [inline src/File.hs:39]: Huh, I guess everyone loves dispatchers.

File: src/File.hs [hunk 2]
```diff
@@ -58,6 +70,9 @@ _extractFile filepath = do
   handler <- openFile filepath ReadMode
   hSetEncoding handler latin1
   contents <- hGetContents handler
+  -- Forcing strict evaluation from 
+  --   https://stackoverflow.com/questions/296792/haskell-io-and-closing-files
+  reduceToNormal contents `seq` hClose handler
```
Comment (Julian-Mentasti) [inline src/File.hs:75]: always forgetting to clean after myself.

File: src/File.hs [hunk 3]
```diff
@@ -146,3 +161,13 @@ _extractPaths filepath = do
...
         else map ((filepath ++ "/") ++) contents
   return fullpaths
+
+{-| 
+    @brief Traverses the given list. This forces reading on lazy IO.
+    See here https://stackoverflow.com/questions/296792/haskell-io-and-closing-files for more information.
+    @param list
+    @returns empty tuple
+-}
+reduceToNormal :: [a] -> ()
```
Comment (Julian-Mentasti) [inline src/File.hs:171]: Huh clever

File: src/File.hs (hunk 1 above)
Comment (Julian-Mentasti) [reply src/File.hs:39]: I guess because conditionals in haskell are gross

**Review by Julian-Mentasti — APPROVED** [review]
LGTM

File: src/File.hs (hunk 1 above)
Comment (reggiemcdonald) [reply src/File.hs:39]: Yea it just seems cleaner this way

**Pull request merged**